modular/gater/metadata_handler.go

```diff
@@ -504,3 +506,107 @@ func (g *GateModular) getGroupListHandler(w http.ResponseWriter, r *http.Request
 	w.Header().Set(ContentTypeHeader, ContentTypeJSONHeaderValue)
 	w.Write(b.Bytes())
 }
+
+// listObjectsByObjectIDHandler list objects by object ids
+func (g *GateModular) listObjectsByObjectIDHandler(w http.ResponseWriter, r *http.Request) {
+	var (
+		err       error
+		b         bytes.Buffer
+		objects   []*types.Object
+		objectIDs bsdb.ObjectIDs
+		reqCtx    *RequestContext
+	)
+
+	defer func() {
+		reqCtx.Cancel()
+		if err != nil {
+			reqCtx.SetError(gfsperrors.MakeGfSpError(err))
+			log.CtxErrorw(reqCtx.Context(), "failed to list objects by ids", reqCtx.String())
+			MakeErrorResponse(w, err)
+		}
+	}()
+
+	reqCtx, _ = NewRequestContext(r, g)
+
+	err = json.NewDecoder(r.Body).Decode(&objectIDs)
+	if err != nil {
+		log.Errorf("failed to parse object ids", "error", err)
+		err = ErrInvalidQuery
+		return
+	}
+
+	if len(objectIDs.IDs) == 0 {
```

Comment:
How many input IDs shall we allow at most?  Can we have a limit?

Reply:
It has been fixed, and is currently set to a limit of 1000